### PR TITLE
Feature/39523

### PR DESCRIPTION
**Pull request by UrvashiChaudhary**
Create a folder controllers inside src
Create folder user and trainee inside controllers.
Create following files in both folder
Controller.ts
routes.ts
index.ts

Define following routes in the trainee
get - /
post - /
put - /
delete - /

Define a class in Controller file having a method for each route returning some fake response.
Create a router.ts file in the root of src directory and import all the created routes and export them.
Import the router to the server file from router.ts file and use them in setUpRoutes method with /api prefix.

### CODE REVIEW

File: src/controllers/trainee/Controller.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    static getInstance(){
+        if(TraineeController.instance){
+            return TraineeController.instance
+        }
+
+        TraineeController.instance = new TraineeController();
+        return TraineeController.instance;
+    }
+    get(req, res, next){
```
Comment (adilsuccessive) [inline src/controllers/trainee/Controller.ts]: type of req, res and next missing.

File: src/controllers/trainee/Controller.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                    name: "Trainee",
+                    address: "Noida"
+                }
+            ]
+            })
+        } catch(err){
+                console.log("Inside err",err);
+        }
+    }
+    create(req, res, next){
```
Comment (adilsuccessive) [inline src/controllers/trainee/Controller.ts]: type of req, res, next missing.

File: src/controllers/trainee/Controller.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    name: "Trainee1",
+                    address: "Noida"
+                }
+            ]
+            })
+        } catch(err){
+                console.log("Inside err",err);
+        }
+    }
+    update(req, res, next){
```
Comment (adilsuccessive) [inline src/controllers/trainee/Controller.ts]: type missing.

File: src/controllers/trainee/Controller.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    name: "Trainee2",
+                    address: "Noida"
+                }
+            ]
+            })
+        } catch(err){
+                console.log("Inside err",err);
+        }
+    }
+    delete(req, res, next){
```
Comment (adilsuccessive) [inline src/controllers/trainee/Controller.ts]: type missng.

File: src/router.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { Router } from 'express';
+import { traineeRouter } from './controllers';
+
+const mainRouter =  Router();
```
Comment (adilsuccessive) [inline src/router.ts]: type of mainRouter is missing.

File: src/controllers/trainee/routes.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { Router } from 'express';
+import traineeController from './Controller'
+
+import validationHandler from '../../libs/validationHandler';
+
+import validation from './validation'
+
+const traineeRouter = Router();
```
Comment (adilsuccessive) [inline src/controllers/trainee/routes.ts]: type of traineeRouter is missing.

**Review by adilsuccessive — CHANGES_REQUESTED** [review]
Fix all lint issues

File: src/controllers/trainee/Controller.ts (hunk 1 above)
Comment (UrvashiChaudhary) [reply src/controllers/trainee/Controller.ts]: done

File: src/controllers/trainee/Controller.ts (hunk 2 above)
Comment (UrvashiChaudhary) [reply src/controllers/trainee/Controller.ts]: done

File: src/controllers/trainee/Controller.ts (hunk 3 above)
Comment (UrvashiChaudhary) [reply src/controllers/trainee/Controller.ts]: done

File: src/controllers/trainee/Controller.ts (hunk 4 above)
Comment (UrvashiChaudhary) [reply src/controllers/trainee/Controller.ts]: done

File: src/router.ts (hunk 5 above)
Comment (UrvashiChaudhary) [reply src/router.ts]: done

File: src/controllers/trainee/routes.ts (hunk 6 above)
Comment (UrvashiChaudhary) [reply src/controllers/trainee/routes.ts]: done

File: src/controllers/trainee/Controller.ts (hunk 1 above)
Comment (UrvashiChaudhary) [reply src/controllers/trainee/Controller.ts]: lint issues are resolved

**Pull request merged**